### ISSUE TITLE
Bug: [no-meaningless-void-operator] False positive on assignment expression voiding

### ISSUE THREAD

**Issue by dl-pt** [typescript-eslint/typescript-eslint#12852]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.
- [x] If I used AI assistance to prepare this issue, I have reviewed and followed the [AI Contribution Policy](https://typescript-eslint.io/contributing/ai-policy/).

### Playground Link

https://typescript-eslint.io/play/#ts=6.0.2&fileType=.tsx&code=BQSgBAvAfGBuD2BLAJmYAPSYCMIDcAsAFCwCGATmOoUUA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYUACAXAngA4oDG0AlobgLQrzkB2uA9AwPbUC2iAhg4wHMkyZNQBubcgBNqbYtB6420dFETRoyyGAC%2BIPUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkAhqqrAO4BSZlemArugDTgakEl2TQ8AOW6MAwgAt00ANYcwXXoIC%2BITUA&tokens=false

### Repro Code

```TypeScript
() => void (x = 1);
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-meaningless-void-operator": "error",
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

No error reported

### Actual Result

Error reported

### Additional Info

Only noticed after a recent update, so possibly related to https://github.com/typescript-eslint/typescript-eslint/pull/12727

**Comment by devtechedge** [comment]
Working on a fix for the false positive on `void (x = 1)` / assignment expressions. Will add a regression test and open a PR.

**Comment by diptomistry** [comment]
Confirmed the repro and stress-tested several variants (compound assignment, chained assignment, member targets, sequences) — all currently misreport the same way. Planning to fix by allowing AssignmentExpression after unwrap, matching the approach in the earlier (spam-flagged, unreviewed) PR #12855. Also noticed UpdateExpression (++x/x++) has the identical issue but wasn't in the original report — happy to include it in this fix or leave as a follow-up, let me know which you'd prefer.
